File: Detect-Capital/DC.swift
class Solution {
    func detectCapitalUse(_ word: String) -> Bool {
        var flag:Bool?
        guard word.characters.count > 1 else {return true}
        
        for sInd in word.characters.indices {
            let thisChar = String(word[sInd])
            //print(thisChar)

            if sInd == word.startIndex{
                if thisChar.uppercased() != thisChar{
                    flag = false
                }
            }else if sInd == word.index(word.startIndex,offsetBy:1) && flag == nil{
                if thisChar.uppercased() != thisChar{ // low case
                    flag = false
                }else{
                    flag = true
                }
            }else if thisChar.uppercased() != thisChar && flag == true{
                return false
            }else if thisChar.uppercased() == thisChar && flag == false{
                return false
            }
        }
        
        return true
    }
}

let a = Solution()
a.detectCapitalUse("USA") // true
a.detectCapitalUse("FlaG") // false
a.detectCapitalUse("ggg") // true

File: Detect-Capital/DC.rs
fn detect_capital_use(word: &String) -> bool {
    let mut flag = -1;

    for (i, c) in word.chars().enumerate() {
        if i == 0 {
            if c.is_uppercase() {
                flag = 0;
            } else {
                flag = 2;
            }
            continue;
        }

        if i == 1 && flag != 2 {
            if c.is_uppercase() {
                flag = 1;
            } else {
                flag = 2;
            }
            continue;
        }

        if c.is_uppercase() && flag == 1 {
            continue;
        } else if !c.is_uppercase() && flag == 2 {
            continue;
        } else {
            return false;
        }
    }
    return true;
}

fn main() {
    println!("{}", detect_capital_use(&String::from("USA")));
    println!("{}", detect_capital_use(&String::from("FlaG")));
    println!("{}", detect_capital_use(&String::from("ggg")));
}
